pkg/features/kube_features.go

```diff
@@ -931,6 +937,8 @@ var defaultKubernetesFeatureGates = map[featuregate.Feature]featuregate.FeatureS
 
 	DisableKubeletCloudCredentialProviders: {Default: true, PreRelease: featuregate.Beta},
 
+	DisableNodeKubeProxyVersion: {Default: false, PreRelease: featuregate.Alpha},
```

Comment:
I forget - did we decide that this should go thru alpha (off by default) for a cycle or just that it should be disableable?

We have `featuregate.Deprecated` which is more self-documenting.  Usually we would name that as "TheBehaviorBeingDeprecated", with a `Default: false`, so users could set `--feature-gates=TheBehaviorBeingDeprecated=true`.

Do you think that's better here?

Reply:
Actually, I went and re-read https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/feature-gates.md and that makes it sound like `featuregate.Deprecated` is what we use when we decide to abandon a gate (e.g.  Alpha -> Deprecated).

We should make a decision about using this to represent deprecations of existing functionality.  @liggitt I think it was you who pointed me at this in the first place?

If so, I can send a PR to add this to the doc.

Reply:
> We should make a decision about using this to represent deprecations of existing functionality. @liggitt I think it was you who pointed me at this in the first place?

The original intent of that gate state (https://github.com/kubernetes/kubernetes/pull/58878) was for things you could opt back into that were deprecated / on their way out.

Reply:
SGTM https://github.com/kubernetes/community/pull/7583

Reply:
> I forget - did we decide that this should go thru alpha (off by default) for a cycle or just that it should be disableable?

I hadn't known about `featuregate.Deprecated` when we started the KEP so I'd suggested the standard alpha, beta, GA.

cloud-provider-gcp in 1.28 still expects KubeProxyVersion to be set, so having kubelet 1.29 not set it by default would result in bumpy upgrades. So we either need to do an alpha, or else wait for 1.30 to do it as a deprecation.

Reply:
ok